marklogic-client-api/src/test/java/com/marklogic/client/test/rows/VectorTest.java

```diff
@@ -0,0 +1,155 @@
+package com.marklogic.client.test.rows;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.marklogic.client.FailedRequestException;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.row.RowRecord;
+import com.marklogic.client.type.ServerExpression;
+import com.marklogic.client.type.XsDoubleSeqVal;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class VectorTest extends AbstractOpticUpdateTest {
+
+	XsDoubleSeqVal sampleVector;
+	XsDoubleSeqVal twoDimensionalVector;
+	ServerExpression invalidVector;
+
+	@BeforeEach
+	void beforeEach() {
+		sampleVector = op.xs.doubleSeq(1.2, 3.4, 5.6);
+		twoDimensionalVector = op.xs.doubleSeq(1.2, 3.4);
+		invalidVector = op.xs.string("InvalidVector");
+		rowManager.withUpdate(false);
+	}
+
+	@Test
+	void cosineSimilarity() {
+		PlanBuilder.ModifyPlan plan =
+			op.fromView("vectors", "persons")
+				.bind(op.as("sampleVector", sampleVector))
+				.bind(op.as("summaryCosineSim", op.vec.cosineSimilarity(op.col("embedding"),op.col("sampleVector"))))
+				.select(op.col("name"), op.col("summary"), op.col("summaryCosineSim"))
+          		.orderBy(op.desc(op.col("summaryCosineSim")))
+				.limit(5);
+		List<RowRecord> rows = resultRows(plan);
+		assertEquals(2, rows.size());
+		rows.forEach(row -> {
+			// Simply a sanity check that the Cosine Similarity score is valid.
+			Assertions.assertTrue(
```

Comment:
For all the "assertTrue" assertions - without a message, if any of these fail, it's usually impossible to know what the actual value is; JUnit won't show anything. And with this being brand new stuff - and thus more likely to change - I think we'll save time in the future by improving these assertions to include a message along the lines of "Unexpected value: " + something - i.e.

```
double val = row.getDouble("summaryCosineSim");
assertTrue(val > 0 && val < 1, "Unexpected value: " + value);
```

Note that you don't need "Assertions." either since you've already statically imported assertTrue.